.github/workflows/gnovm.yml

```diff
@@ -62,22 +62,22 @@ jobs:
     runs-on: ubuntu-latest
     timeout-minutes: 15
     steps:
+      - uses: actions/checkout@v4
       - uses: actions/setup-go@v4
         with:
-          go-version: ${{ matrix.go-version }}
-      - uses: actions/checkout@v4
+          go-version: ${{ matrix.goversion }}
       - name: test
         working-directory: gnovm
         run: |
           export GOPATH=$HOME/go
           export GOTEST_FLAGS="-v -p 1 -timeout=30m -coverprofile=coverage.out -covermode=atomic"
           make ${{ matrix.args }}
-      - if: runner.os == 'Linux'
+      - if: ${{ runner.os == 'Linux' && matrix.goversion == '1.21.x' }}
```

Comment:
this might be a problem when we update go versions and 1.21 is not supported anymore, we can lose coverage execution

Reply:
yeah, I'm the one who made the 3 last go version bumps and I basically make sure to update all the references I can find.

- go1.21: https://github.com/gnolang/gno/pull/1093
- go1.20: https://github.com/gnolang/gno/pull/518
- go1.19: https://github.com/gnolang/gno/pull/310

at some point, I was considering adding a variable.yml file with "latest_go" and "previous_go" variable; or maybe we can add a quick misc/check-go-versions.sh script to assist us or to check in the CI.